### PR TITLE
Feature/clc 673

### PR DESCRIPTION
**Pull request by echistyakov**

### CODE REVIEW

File: Cielo24/Enums.cs [hunk 1]
```diff
@@ -1,217 +1,249 @@
...
+        JobCreated,
+        JobDeleted,
+        JobAddMedia,
+        JobAddTranscript,
+        JobPerformTranscription,
+        JobPerformPremiumSync,
+        JobUpdateElementlist,
+        JobGetTranscript,
+        JobGetCaption,
+        JobGetElementlist
```
Comment (echistyakov) [inline Cielo24/Enums.cs]: All of the above enums need to have a Description attribute in order to be comparable against strings of the form HELLO_WORLD (returned by our API). Not sure if enums is the best approach to this problem, however. Maybe a static class with public string constants would be more suitable?

File: Cielo24/Enums.cs [hunk 2]
```diff
@@ -1,217 +1,249 @@
...
+        InvalidQuery,
+        InvalidOption,
+        InvalidUrl,
+        MissingParameter,
+        NotImplemented,
+        ItemNotFound,
+        InvalidReturnHandlers,
+        NotParentAccount,
+        NoChildrenFound,
+        UnhandledError
```
Comment (echistyakov) [inline Cielo24/Enums.cs]: Same as above.

File: Cielo24/Enums.cs [hunk 3]
```diff
@@ -1,217 +1,249 @@
...
+    public enum JobPriority
     {
-        ECONOMY,
-        STANDARD,
-        PRIORITY,
-        CRITICAL
+        Economy,
+        Standard,
+        Priority,
+        Critical
```
Comment (echistyakov) [inline Cielo24/Enums.cs]: Is Newtonsoft.JSON library case insensitive when it comes to parsing strings to enums?
If not, each of the above enums will need an all-caps Description attribute as well as  `[JsonConverter(typeof(DescriptionToEnumConverter))]`

File: Cielo24/Enums.cs [hunk 4]
```diff
@@ -248,22 +280,14 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
...
-                }
-                else
-                {
-                    return base.ReadJson(reader, objectType, existingValue, serializer);
-                }
+                return Fidelity.Premium;
             }
+            return readerValue.Equals("HIGH")
+                ? Fidelity.Professional
+                : base.ReadJson(reader, objectType, existingValue, serializer);
```
Comment (echistyakov) [inline Cielo24/Enums.cs]: This makes the code shorter, but a little less clear. Can we replace this with if-elseif-else?
```
if STANDARD:
   ....
else if HIGH:
   ....
else:
   ....
```

**Comment by aerosol** [comment]
:heart_eyes:

File: Cielo24/Enums.cs (hunk 1 above)
Comment (gornik) [inline Cielo24/Enums.cs]: Replace with `public string const XXX`

File: Cielo24/Enums.cs (hunk 2 above)
Comment (gornik) [inline Cielo24/Enums.cs]: Added `Description` attributes and used this enum in the exception.

File: Cielo24/Enums.cs (hunk 3 above)
Comment (gornik) [inline Cielo24/Enums.cs]: Yes, it's case insensitive. Also changed the `GetDescription` extension method to take this into account.

File: Cielo24/Cielo24.csproj [hunk 5]
```diff
@@ -68,20 +69,14 @@
...
     <Compile Include="Options\TranscriptionOptions.cs" />
+    <Compile Include="Paths.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="QueryBuilder.cs" />
     <Compile Include="Utils.cs" />
     <Compile Include="WebUtils.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="packages.config" />
-    <None Include="_CreateNewNuGetPackage\Config.ps1" />
```
Comment (echistyakov) [inline Cielo24/Cielo24.csproj]: Why was the following removed? It assembles the DLL automatically, and generates a NuGet.

File: Cielo24/Enums.cs [hunk 6]
```diff
@@ -1,217 +1,265 @@
...
+        Good,
         [Description("Bad")]
-        BAD,
+        Bad,
         [Description("Unknown")]
-        UNKNOWN
+        Unknown
     }
 
+    [Flags]
```
Comment (echistyakov) [inline Cielo24/Enums.cs]: Nice!

**Pull request closed without merging**